### ISSUE TITLE
obj meshes seem to randomly loose material override settings

### ISSUE THREAD

**Issue by AndHalfAFish** [godotengine/godot#86665]
### Tested versions

Godot 4.1.3.stable

### System information

Godot v4.1.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4080 Laptop GPU (NVIDIA; 31.0.15.3713) - 13th Gen Intel(R) Core(TM) i7-13700HX (24 Threads)

### Issue description

Obj meshes randomly loose material override setting. Doesn't matter if the mesh uses 1 or 2 or more materials. Similar meshes behave differently. Only happens w/ obj, not gltf.

![obj_material_override](https://github.com/godotengine/godot/assets/59080934/e7df28e2-4511-40f1-9026-8b74be992d3b)

### Steps to reproduce

obj mesh dragged into scene , add surface material override.
Sometimes the material override will be gone after re-opening the editor, sometimes the material override will remain set.

### Minimal reproduction project (MRP)

n.a. : seems random.

**Comment by RandomCatDude** [comment]
i believe this happens when the `.obj` is reimported. i've dealt with this in my project and it consistently happens when i update the `.obj` and godot reimports it.

also; should clarify that this is specific to surface material overrides. regular material override is unaffected.

**Comment by AndHalfAFish** [comment]
Good to know it's not me doing something wrong. =)
In my case, it happened without reimport of the .obj or updating the override material. 
Not sure if it is relevant : the scene containing the .obj + override materials is 120kb / 2450 lines.

**Comment by AndHalfAFish** [comment]
More .obj - override material information : 

In editor :
![obj_in_editor](https://github.com/godotengine/godot/assets/59080934/399ee70a-24bd-41cc-8444-78bc6f21f0f3)

in game : 
![obj_in_game](https://github.com/godotengine/godot/assets/59080934/15840c2c-24da-4002-bf5c-abfad3ab8cb0)

After project reload, picture frames have lost their override material (look like 2nd image).

**Comment by RandomCatDude** [comment]
yeah the editor just seems to erroneously not update when the overrides are lost
if you reload the scene in any way without saving it (running the project saves every scene), the overrides are restored

**Comment by actually-reb** [comment]
Has this issue been looked into? This bug makes it very annoying to work with meshes, as this happens to me with all mesh resources, including mesh resources saved from .gltf files. The only workaround I've found to be able to both use Godot materials and be able to reimport meshes freely is to use a .gltf scene directly, and give it editable children.
